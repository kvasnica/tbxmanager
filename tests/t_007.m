function t_007
% tbxmanager uninstall

onCleanup(@(x) tbx_setupTest('done')); tbx_setupTest('start');

% set up the test repo
tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test.xml
tbxmanager so re http://control.ee.ethz.ch/~mpt/tbx/ifa.xml
tbxmanager show sources

% install two packages
tbxmanager install tbx1 tbx2 tbx3 tbx4
tbxmanager show installed

% uninstall some
tbxmanager uninstall tbx3 tbx4
tbxmanager show installed

% add a new repo
tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test_update.xml

% update all should update tbx1, tbx2, tbx3
tbxmanager update tbx1 tbx2
% uninstall tbx2
tbxmanager uninstall tbx2

end
