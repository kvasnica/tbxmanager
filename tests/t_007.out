--- tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test.xml 
--- tbxmanager so re http://www.tbxmanager.com/package/index.xml 
--- tbxmanager show sources 
Active sources:

http://control.ee.ethz.ch/~mpt/tbx/test/test.xml
--- tbxmanager install tbx1 tbx2 tbx3 tbx4 

Installing version "1.0" of "tbx1"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx1_v1.zip"...
"tbx1:1.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx1/1.0/all
Toolbox "tbx1:1.0:all" added to the Matlab path.

Installing version "1.0" of "tbx2"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx2_v1.zip"...
"tbx2:1.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/1.0/all
Toolbox "tbx2:1.0:all" added to the Matlab path.

Installing version "2.0" of "tbx3"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx3_v2.tgz"...
"tbx3:2.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx3/2.0/all
Toolbox "tbx3:2.0:all" added to the Matlab path.

Installing version "1.0" of "tbx4"...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx4.tgz"...
"tbx4:1.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx4/1.0/all
Toolbox "tbx4:1.0:all" added to the Matlab path.
--- tbxmanager show installed 
Locally installed packages:

tbx1   Version 1.0       (@current_date@)
tbx2   Version 1.0       (@current_date@)
tbx3   Version 2.0       (@current_date@)
tbx4   Version 1.0       (@current_date@)
--- tbxmanager uninstall tbx3 tbx4 

Toolbox "tbx3:2.0:all" removed from the Matlab path.
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx3/2.0/all"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx3/2.0"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx3"...
Toolbox "tbx3" uninstalled.

Toolbox "tbx4:1.0:all" removed from the Matlab path.
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx4/1.0/all"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx4/1.0"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx4"...
Toolbox "tbx4" uninstalled.
--- tbxmanager show installed 
Locally installed packages:

tbx1   Version 1.0       (@current_date@)
tbx2   Version 1.0       (@current_date@)
--- tbxmanager source add http://control.ee.ethz.ch/~mpt/tbx/test/test_update.xml 
--- tbxmanager update tbx1 tbx2 
Toolbox "tbx1" has new version "2.0", installing...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx1_v2.zip"...
"tbx1:2.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx1/2.0/all
Toolbox "tbx1:2.0:all" added to the Matlab path.
Toolbox "tbx2" has new version "2.0", installing...
Downloading "http://control.ee.ethz.ch/~mpt/tbx/test/tbx2_v2.zip"...
"tbx2:2.0:all" installed to /Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/2.0/all
Toolbox "tbx2:2.0:all" added to the Matlab path.
--- tbxmanager uninstall tbx2 

Toolbox "tbx2:1.0:all" removed from the Matlab path.
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/1.0/all"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/1.0"...
Toolbox "tbx2" uninstalled.

Toolbox "tbx2:2.0:all" removed from the Matlab path.
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/2.0/all"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2/2.0"...
Removing directory "/Users/michal/matlab/mpt3/hg/tbxmgr-mk/tests/tbxstorage_test/tbx2"...
Toolbox "tbx2" uninstalled.
